AutoRaise.mm

```diff
@@ -475,48 +475,140 @@ CGEventRef eventTapHandler(CGEventTapProxy proxy, CGEventType type, CGEventRef e
+        [[NSUserDefaults standardUserDefaults] registerDefaults:standardDefaults];
+        
+        // put initial values to our variables
+        delayCount = (int) [[NSUserDefaults standardUserDefaults] integerForKey:defaultDelay];
+        warpX = (float) [[NSUserDefaults standardUserDefaults] floatForKey:defaultWarpX];
+        warpY = (float) [[NSUserDefaults standardUserDefaults] floatForKey:defaultWarpY];
+        cursorScale = (float) [[NSUserDefaults standardUserDefaults] floatForKey:defaultScale];
+
+        // if there are any arguments, we would use only them
+        if (argc > 1) {
+            // but if -delay argument not valid, we would exit
+            if (not (delayCount > 0)) {
+                fprintf(stderr, "No valid arguments\nRun without any arguments to use default settings");
+                return 0;
             }
+            // check is there valid warps
+            if (warpX >= 0 and warpX <= 1 and warpY >= 0 and warpY <= 1) { warpMouse = true; }
         } else {
+            // we are looking for config files
             NSString * home = NSHomeDirectory();
-            NSFileHandle * delayFile = [NSFileHandle fileHandleForReadingAtPath:
-                [NSString stringWithFormat: @"%@/AutoRaise.delay", home]];
-            if (delayFile) {
-                delayCount = abs([[[NSString alloc] initWithData:
-                    [delayFile readDataOfLength: 2] encoding:
-                    NSUTF8StringEncoding] intValue]);
-                [delayFile closeFile];
-            }
-            NSFileHandle * warpFile = [NSFileHandle fileHandleForReadingAtPath:
-                [NSString stringWithFormat: @"%@/AutoRaise.warp", home]];
-            if (warpFile) {
-                warpMouse = true;
-                NSString * line = [[NSString alloc] initWithData:
-                    [warpFile readDataOfLength:11] encoding:
-                    NSUTF8StringEncoding];
-                NSArray * components = [line componentsSeparatedByString: @" "];
-                if (components.count >= 1) { warpX = [[components objectAtIndex:0] floatValue]; }
-                if (components.count >= 2) { warpY = [[components objectAtIndex:1] floatValue]; }
-                if (components.count >= 3) { cursorScale = [[components objectAtIndex:2] floatValue]; }
-                [warpFile closeFile];
+            
+            NSString * oldStyleDelayFile   = [NSString stringWithFormat: @"%@/AutoRaise.delay",          home];
+            NSString * oldStyleWarpFile    = [NSString stringWithFormat: @"%@/AutoRaise.warp",           home];
+            NSString * newStyleConfigFile1 = [NSString stringWithFormat: @"%@/.AutoRaise",               home];
+            NSString * newStyleConfigFile2 = [NSString stringWithFormat: @"%@/.config/AutoRaise/config", home];
+            
+            #define isFileExist(string) [[NSFileManager defaultManager] fileExistsAtPath:string]
+            
+            // old style config files
```

Comment:
I would move the old and new style code into their own methods, just to keep things more understandable.

Reply:
Ok!